dao/src/test/java/com/iluwatar/dao/CustomerDaoImplTest.java

```diff
@@ -0,0 +1,81 @@
+package com.iluwatar.dao;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class CustomerDaoImplTest {
+
+	private CustomerDaoImpl impl;
+	private List<Customer> customers;
+	private static final Customer CUSTOMER = new Customer(1, "Freddy", "Kruger");
+	
+	@Before
+	public void setUp() {
+		customers = new ArrayList<Customer>();
+		customers.add(CUSTOMER);
+		impl = new CustomerDaoImpl(customers);
+	}
+	
+	@Test
+	public void deleteExistingCustomer() {
+		assertEquals(1, impl.getAllCustomers().size());
```

Comment:
Suggestion: If we are improving tests why not go a step further and add junit-hamcrest dependency as well. It will make this assertion read like
`assertThat(impl.getAllCustomers().size(), equals(1));`